kotlin-yaml/src/jsMain/kotlin/yaml/parse.kt

```diff
@@ -0,0 +1,19 @@
+package yaml
+
+@JsModule("yaml")
+external object YAML {
+    /**
+     * Parse an input string into JavaScript.
+     *
+     * Only supports input consisting of a single YAML document; for multi-document
+     * support you should use `YAML.parseAllDocuments`. May throw on error, and may
+     * log warnings using `console.warn`.
+     *
+     * @param str - A string with YAML formatting.
+     * @param reviver - A reviver function, as in `JSON.parse()`
+     * @returns The value will match the type of the root value of the parsed YAML
+     *   document, so Maps become objects, Sequences arrays, and scalars result in
+     *   nulls, booleans, numbers and strings.
+     */
+    fun <T> parse(src: String): T
```

Comment:
Non-null `T`?

Reply:
`T` can be a nullable type, so does it matter?

Reply:
Hm, actually I think `Any?` might be better, otherwise you have to type the call.
With `T` I cannot just do `yaml.parse("null")` or `val foo = yaml.parse("null")` but have to do `val foo: Any? = yaml.parse("null")`.
On the other hand with `Any?` then have to cast if you do want to assign it somewhere typed.
What do you think?

Reply:
Ah, ok, judging by `JSON.parse` it seems you prefer it like this with `T` :-)

Reply:
`JSON.parse` also can return nullable values :(

Reply:
Yes, indeed interestingly it just works even with the old signature.
But I guess removing the `: Any` would be appropriate.